### PR TITLE
TOML documentation update - User Guide

### PR DESCRIPTION
**Pull request by janciesla8818**
This PR updates documentation to reflect the new configuration format and accounts to updated documentation structure in:
* User Guide section
* Tutorials section
* Updates broken links to listen section 
* minor typos and broken links fixes

### CODE REVIEW

File: doc/user-guide/Getting-started.md [hunk 1]
```diff
@@ -89,9 +89,7 @@ mongooseimctl bootstrap
 
 It executes scripts inside the `scripts/` directory with a `bootstrap` prefix in alphabetical order. [More information](Bootstrap-Scripts.md)
 
-Execute /Users/mikhailuvarov/erlang/esl/MongooseIM/_build/prod/rel/mongooseim/scripts/bootstrap01-hello.sh
-Hello from /Users/mikhailuvarov/erlang/esl/MongooseIM/_build/prod/rel/mongooseim/scripts/bootstrap01-hello.sh script.
-MongooseIM is installed into /Users/mikhailuvarov/erlang/esl/MongooseIM/_build/prod/rel/mongooseim
+Execute `Hello` from `scripts/bootstrap01-hello.sh` script you can find in the release directory `$REPO_DIR/_build/prod/rel/mongooseim`.
```
Comment (NelsonVides) [inline doc/user-guide/Getting-started.md:92]:
```suggestion
Execute `Hello` from the `scripts/bootstrap01-hello.sh` script that you can find in the release directory `$REPO_DIR/_build/prod/rel/mongooseim`.
```

File: doc/user-guide/client-certificate.md [hunk 2]
```diff
@@ -5,47 +5,51 @@ This method uses the `SASL EXTERNAL` mechanism.
 
 ## Server-side prerequisites
 
-### Properly configure `ejabberd_c2s` listener
+### Properly configure Client-to-server (C2S) listener
 
 A server must request the certificate from a client, so you'll need to enable `verify_peer` option and provide a path to CA chain that may be used for client's certificate check (`cafile` option).
 
-Please check [Listener modules](../advanced-configuration/Listener-modules.md#client-to-server-c2s-ejabberd_c2s) page for more information or simply follow the examples at the end of this section.
+Please check [Listener modules](../../advanced-configuration/listen/#client-to-server-c2s-listenc2s) page for more information or simply follow the examples at the end of this section.
```
Comment (NelsonVides) [inline doc/user-guide/client-certificate.md:12]:
```suggestion
Please check the [Listener modules](../../advanced-configuration/listen/#client-to-server-c2s-listenc2s) page for more information or simply follow the examples at the end of this section.
```

**Comment by mongoose-im** [comment]
[8614.1](https://travis-ci.org/esl/MongooseIM/jobs/732221442) / Erlang 23.0.3 / small_tests / b26eab1ddfa29988370f8f8a5b96f89cb7bd2d75
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2898/8614/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2898/8614/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2020-10-02_11.24.38/lib.mongooseim.logs/run.2020-10-02_11.24.38/suite.log.html)

File: doc/user-guide/client-certificate.md [hunk 3]
```diff
@@ -5,47 +5,51 @@ This method uses the `SASL EXTERNAL` mechanism.
...
 
-### Properly configure `ejabberd_cowboy` listener
+### Properly configure `http` listener
 
 SASL EXTERNAL authentication is also possible for WebSocketSecure and BOSH connections over HTTPS.
-Similarly as in `ejabberd_c2s` case, the server must request the certificate from the client.
-In this case it's enabled by adding the following options to `ssl` option of `ejabberd_cowboy` :
+Similarly as in `client-to-server` case, the server must request the certificate from the client.
+In this case it's enabled by adding the following options to `tls` option of `listen.http` :
```
Comment (NelsonVides) [inline doc/user-guide/client-certificate.md:18]:
```suggestion
Similarly as in the `client-to-server` case, the server must request the certificate from the client.
In this case it's enabled by adding the following options to the `tls` option of `listen.http`:
```

File: doc/user-guide/client-certificate.md [hunk 4]
```diff
@@ -5,47 +5,51 @@ This method uses the `SASL EXTERNAL` mechanism.
...
-Its value must include a `cyrsasl_external` item.
 Obviously the list may be longer, if the system should support both the certificate and password based authentication.
 
-The `SASL EXTERNAL` authentication method requires a digital client certificate.
+The `SASL EXTERNAL` authentication mechanism requires a digital client certificate.
 This digital certificate should contain `xmppAddr` field(s), which is always checked first.
 If there is more than one JID specified in the `xmppAddr` fields, the client must include the authorisation entity which corresponds to the one of the specified JIDs.
 
-When no `xmppAddr` is specified, the `cn` (common name) field might be used to provide client's username, but it is optional and can be configured with the [`sasl_external`](../advanced-configuration/auth.md#auth-sasl-external) option in the `auth` section.
+When no `xmppAddr` is specified, the `cn` (common name) field might be used to provide client's username, but it is optional and can be configured with the [`sasl_external`](../advanced-configuration/auth.md#authsasl_external) option in the `auth` section.
```
Comment (NelsonVides) [inline doc/user-guide/client-certificate.md:40]:
```suggestion
When no `xmppAddr` is specified, the `cn` (common name) field might be used to provide the client's username, but it is optional and can be configured with the [`sasl_external`](../advanced-configuration/auth.md#authsasl_external) option in the `auth` section.
```

File: doc/user-guide/Get-to-know-MongooseIM.md [hunk 5]
```diff
@@ -32,15 +32,19 @@ MongooseIM can be deployed for a number of scenarios fitting customer needs. The
 
 A more scalable solution would be deploying MongooseIM with an external database for persistent data. Such a setup requires a cluster of MongooseIM nodes, an external database, and a load balancer to manage the traffic from the client applications.
 
-If deployed on a 16 GB RAM machine with at least 4 cores, a single MongooseIM node can handle 200-300 thousands of online users. This setup is suitable for systems with up to 10 nodes.
+A single MongooseIM node can handle 2.5 million online users.
```
Comment (NelsonVides) [inline doc/user-guide/Get-to-know-MongooseIM.md:35]:
```suggestion
A single MongooseIM node can handle as many as 2.5 million online users.
```

File: doc/user-guide/ICE_tutorial.md [hunk 6]
```diff
@@ -61,9 +61,9 @@ In order to enable signalling we need an instance of [MongooseIM] running with t
 You can find MongooseIM installation instructions on [this page](Getting-started.md).
 Once you have cloned the repository and compiled the project, you need to modify the `mongooseim.cfg` config file (you can find this file at `$REPO/_build/prod/rel/mongooseim/etc/mongooseim.cfg`, where `$REPO` is a top-level directory of the cloned repo).
```
Comment (gustawlippa) [inline doc/user-guide/ICE_tutorial.md:62]: A pesky `mongooseim.cfg`

File: doc/user-guide/ICE_tutorial.md [hunk 7]
```diff
@@ -61,9 +61,9 @@ In order to enable signalling we need an instance of [MongooseIM] running with t
...
 You can use [this](ICE_tutorial/mongooseim.cfg) configuration file and modify the relevant part:
```
Comment (gustawlippa) [inline doc/user-guide/ICE_tutorial.md:63]: This file is in the `doc/user-guide/ICE_tutorial/` directory and seems old and is not TOML for sure, so the modification described below will not work. I think we should either provide a new file or just send the user to the default that MIM contains in `rel/files`?

File: doc/user-guide/ICE_tutorial.md (hunk 6 above)
Comment (janciesla8818) [reply doc/user-guide/ICE_tutorial.md:62]: Good catch!

File: doc/user-guide/ICE_tutorial.md (hunk 7 above)
Comment (janciesla8818) [reply doc/user-guide/ICE_tutorial.md:63]: I will remove this sentence from line 63 and I already created a ticket to provide the user with sample TOML config.

**Review by gustawlippa — APPROVED** [review]
Looks great!
Not directly connected to this PR, but I'd remove a stray `"` from `Features-and-supported-standards.md`, remove reference to MIM 2.2 from `MongooseIM-High-level-Architecture/#services`, and change "Contact us." from `MongooseIM-High-level-Architecture/#mongooseim-in-a-worldwide-multi-datacenter-configuration` to something more polished.
But these are just little unrelated stylistic things and generally the PR is great, thanks for the work!

**Review by NelsonVides — APPROVED** [review]
Thanks a lot guys for the rework here! 😄

**Pull request merged**